libs/cdk/src/lib/template-management/template-manager.ts

```diff
@@ -0,0 +1,168 @@
+  notificationToTemplateName: {
+    [rxKind: string]: (
+      value: T,
+      templates: { get: (name: N) => TemplateRef<C> }
+    ) => N;
+  };
+}): TemplateManager<T, C, N> {
+  const {
+    renderSettings,
+    notificationToTemplateName,
+    templateSettings,
+  } = config;
+  const {
+    defaultStrategyName,
+    strategies,
+    cdRef: injectingViewCdRef,
+    patchZone,
+    parent,
+    eRef,
+  } = renderSettings;
+
+  const tNode: TNode = parent
+    ? getTNode(injectingViewCdRef, eRef.nativeElement)
+    : false;
+
+  let activeTemplate: N;
+
+  const strategyHandling$ = strategyHandling(defaultStrategyName, strategies);
+  const templates = templateHandling<N, C>();
+  const viewContainerRef = templateSettings.viewContainerRef;
+
+  const createEmbeddedView = getEmbeddedViewCreator(
+    viewContainerRef,
+    patchZone
+  );
+  const triggerHandling = templateTriggerHandling();
+  const getContext = notificationKindToViewContext(
+    templateSettings.customContext
+  );
+
+  return {
+    addTemplateRef: templates.add,
+    addTrigger: triggerHandling.next,
+    nextStrategy: strategyHandling$.next,
+    render(values$: Observable<T>): Observable<any> {
+      return values$.pipe(
+        map((o) => (isObservable(o) ? o : of(o))),
+        distinctUntilChanged(),
+        switchAll(),
+        distinctUntilChanged(),
+        rxMaterialize(),
+        mergeWith(triggerHandling.trigger$ || EMPTY),
+        withLatestFrom(strategyHandling$.strategy$),
+        // Cancel old renders
+        switchMap(([notification, strategy]) => {
+          const kind: RxNotificationKind = notification.kind;
+          const value: T = notification.value as T;
+          const templateName = notificationToTemplateName[kind](
+            value,
+            templates
+          );
+
+          const template = templates.get(templateName);
+          const isNewTemplate = activeTemplate !== templateName;
+          const notifyParent = isNewTemplate || parent;
+          return combineLatest([
+            onStrategy(
+              value,
+              strategy,
+              (v: T, work: RenderWork, options: CoalescingOptions) => {
+                if (isNewTemplate) {
+                  if (viewContainerRef.length > 0) {
+                    viewContainerRef.clear();
+                  }
+                  if (template) {
+                    createEmbeddedView(template);
+                  }
+                }
+                if (template) {
+                  const context = getContext[kind](value);
+                  const view = <EmbeddedViewRef<C>>viewContainerRef.get(0);
+                  Object.keys(context).forEach((k) => {
+                    view.context[k] = context[k];
+                  });
+                  work(view, options.scope, notification);
+                }
+                activeTemplate = templateName;
+              }
+              // @TODO specify scoping
```

Comment:
so did we specify it? :D

Reply:
If there is no scoping I would like to comment why.

Reply:
the problem here is that the template-manager potentially has multiple scopes (1 for each `EmbeddedViewRef` and 1 for the `Component` itself).

in a perfect world we would scope each operation to the template `EmbeddedViewRef` when possible. When creating the first `EmbeddedViewRef` it won't be possible, since the scope gets created in the function which we would like to scope with it... ? :D

I don't even know if we really need to scope to a single `EmbeddedView`

Reply:
In `rxLet` there could e multiple sync emissions from some global state or other sources.

Reply:
So could you add a comment or todo so we don't forget to polish here?

Reply:
@hoebbelsB u can also put it here and I move it into code